# Comparisons

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from seaborn import set_style
from datetime import datetime, timedelta
set_style("whitegrid")

In [2]:
AHH = pd.read_csv('data/AHH.csv')
CIB = pd.read_csv('data/CIB.csv')
MCI = pd.read_csv('data/MCI.csv')
ALOT = pd.read_csv('data/ALOT.csv')
EMKR = pd.read_csv('data/EMKR.csv')
HHC = pd.read_csv('data/HHC.csv')
NSC = pd.read_csv('data/NSC.csv')
UTL = pd.read_csv('data/UTL.csv')
sp500 = pd.read_csv('data/SP500.csv')

In [3]:
AHH=AHH[AHH['y']==1]
CIB=CIB[CIB['y']==1]
MCI=MCI[MCI['y']==1]
ALOT=ALOT[ALOT['y']==1]
EMKR=EMKR[EMKR['y']==1]
HHC=HHC[HHC['y']==1]
NSC=NSC[NSC['y']==1]
UTL=UTL[UTL['y']==1]
sp500=sp500[sp500['y']==1]

In [4]:
stocks = [AHH,CIB,MCI,ALOT,EMKR,HHC,NSC,UTL,sp500]

In [5]:
for stock in stocks:
    stock.columns = stock.columns.str.strip()

In [12]:
for stock in stocks:
    stock['diff_fcmo'] = (stock['Friday_Close']-stock['Monday_Open'])

In [13]:
AHH

Unnamed: 0 Friday_date Monday_date  Friday_Open  Monday_Open  \
7             7  2019-03-08  2019-03-11        15.26        15.22   
38           38  2019-11-01  2019-11-04        18.80        18.73   
54           54  2020-03-06  2020-03-09        17.50        16.84   
84           84  2020-10-30  2020-11-02         8.58         9.18   
99           99  2021-03-12  2021-03-15        13.89        13.90   
129         129  2021-11-05  2021-11-08        15.14        15.64   
144         144  2022-03-11  2022-03-14        14.96        14.95   
173         173  2022-11-04  2022-11-07        11.29        11.35   
187         187  2023-03-10  2023-03-13        12.44        11.76   
217         217  2023-11-03  2023-11-06        10.40        10.50   
231         231  2024-03-08  2024-03-11        10.70        10.60   
262         262  2024-11-01  2024-11-04        10.93        10.78   

     Friday_High  Monday_High  Friday_Low  Monday_Low  Friday_Close  \
7          15.36        15.45       15.15       15.21         15.17   
38         18.89        18.77       18.63       18.45         18.77   
54         17.88        17.00       17.26       15.34         17.86   
84          9.17         9.24        8.58        8.63          9.01   
99         14.00        13.90       13.80       13.38         13.96   
129        15.63        15.64       15.08       15.19         15.55   
144        15.10        14.98       14.76       14.69         14.82   
173        11.46        11.50       11.09       11.16         11.31   
187        12.45        12.17       11.85       11.71         11.94   
217        10.65        10.57       10.40       10.35         10.52   
231        10.78        10.71       10.53       10.54         10.62   
262        10.97        10.95       10.71       10.74         10.75   

     Monday_Close  Friday_Adj Close  Monday_Adj Close  Friday_Volume  \
7           15.43             10.96             11.15         132000   
38          18.50             14.09             13.89         174800   
54          15.37             13.56             11.67         395400   
84           8.85              7.10              6.97         692800   
99          13.53             11.11             10.77         611100   
129         15.29             12.82             12.61         318900   
144         14.83             12.36             12.37         413100   
173         11.40              9.84              9.92         293700   
187         11.90             10.56             10.52         454200   
217         10.44              9.80              9.72         390100   
231         10.71             10.04             10.13         429600   
262         10.91             10.75             10.91         485100   

     Monday_Volume  y  diff_fcmo  
7           231000  1      -0.05  
38          239600  1       0.04  
54          316000  1       1.02  
84          354900  1      -0.17  
99          320300  1       0.06  
129         272500  1      -0.09  
144         356100  1      -0.13  
173         303300  1      -0.04  
187         481200  1       0.18  
217         225000  1       0.02  
231         325500  1       0.02  
262         609300  1      -0.03

In [24]:
stock_names = ['AHH','CIB','MCI','ALOT','EMKR','HHC','NSC','UTL','sp500']
p_val = [('AHH',0.008972), ('CIB',0.041472), ('MCI',0.005252), ('ALOT',0.041177),('EMKR',0.000603),('HHC',0.043817), ('UTL',0.041104),('UTL',0.049185)]

In [21]:
profit = []

for stock, name in zip(stocks,stock_names):
    stock_count=0
    for i in range(len(stock)):
        # only look at the fall, not spring. 
        if datetime.strptime(stock.iloc[i]['Friday_date'],"%Y-%m-%d").month>=7:
            stock_count+=stock.iloc[i]['diff_fcmo']
    profit.append((name,stock_count))


In [22]:
profit

[('AHH', -0.2699999999999996),
 ('CIB', -0.7499999999999964),
 ('MCI', -0.17999999999999794),
 ('ALOT', 0.34999999999999787),
 ('EMKR', -1.0699999999999996),
 ('HHC', -1.2899999999999778),
 ('NSC', -6.80000000000004),
 ('UTL', 1.289999999999992),
 ('sp500', -54.67999999999938)]

In [27]:
sorted(p_val, key=lambda x: x[1])

[('EMKR', 0.000603),
 ('MCI', 0.005252),
 ('AHH', 0.008972),
 ('UTL', 0.041104),
 ('ALOT', 0.041177),
 ('CIB', 0.041472),
 ('HHC', 0.043817),
 ('UTL', 0.049185)]

In [55]:
sum_inverse = 0
for i in range(len(p_val)):
    print(1/p_val[i][1])
    sum_inverse+=1/p_val[i][1]

111.45786892554614
24.11265432098765
190.4036557501904
24.285402044830853
1658.374792703151
22.822192299792317
24.328532502919423
20.33140185015757


In [56]:
sum_inverse

2076.1165003975752

In [57]:
p_val_norm=[(name,value,(1/value)/sum_inverse) for name, value in p_val]


In [58]:
p_val_norm

[('AHH', 0.008972, 0.05368574880272953),
 ('CIB', 0.041472, 0.011614306960312723),
 ('MCI', 0.005252, 0.09171145054419066),
 ('ALOT', 0.041177, 0.011697514103943691),
 ('EMKR', 0.000603, 0.7987869622853886),
 ('HHC', 0.043817, 0.010992732004885987),
 ('UTL', 0.041104, 0.011718288688645613),
 ('UTL', 0.049185, 0.00979299660990321)]

In [61]:
df=pd.DataFrame(p_val_norm)

In [62]:
df

0         1         2
0   AHH  0.008972  0.053686
1   CIB  0.041472  0.011614
2   MCI  0.005252  0.091711
3  ALOT  0.041177  0.011698
4  EMKR  0.000603  0.798787
5   HHC  0.043817  0.010993
6   UTL  0.041104  0.011718
7   UTL  0.049185  0.009793

In [63]:
df.columns=['ticker','p_val','1/p_val%']

In [64]:
df

ticker     p_val  1/p_val%
0    AHH  0.008972  0.053686
1    CIB  0.041472  0.011614
2    MCI  0.005252  0.091711
3   ALOT  0.041177  0.011698
4   EMKR  0.000603  0.798787
5    HHC  0.043817  0.010993
6    UTL  0.041104  0.011718
7    UTL  0.049185  0.009793

In [65]:
profit

[('AHH', -0.2699999999999996),
 ('CIB', -0.7499999999999964),
 ('MCI', -0.17999999999999794),
 ('ALOT', 0.34999999999999787),
 ('EMKR', -1.0699999999999996),
 ('HHC', -1.2899999999999778),
 ('NSC', -6.80000000000004),
 ('UTL', 1.289999999999992),
 ('sp500', -54.67999999999938)]

In [68]:
combo=0
for i in range(len(profit)-1):
    combo+= p_val_norm[i][2]*profit[i][1]

In [69]:
combo

-0.971553884946229

Summary: I took only the fall DST weekends and took Friday close - Monday open; if we sell on Friday and buy on Monday, this quantity is what we gain/lose. It looks like we usually lose money this way; the stock prices are actually increasing over many of these particular DST weekends from 2019-2024 (including for the S&P 500).

But anyways, for each individual stock, I added those losses/gains to get a list of 8 numbers (since I only found 8 of the 13 stocks that Gabby picked). I then took the p-values for the t-test, took their inverse (since a very small p-value means something really significant because it's super unlikely to see that under the null hypothesis), summed them up, and then got the percentage that each inverse p-value represents. For example, EMKR had a tiny p-value so its inverse p-value represents a large percentage.
I then took a linear combination of the previous list of numbers weighted by these inverse p-values which ends up being -0.97 which is better than the S&P500's -54.68